logs/summary.go

```diff
@@ -95,3 +107,35 @@ func (self *LogSummaryHandler) dumpBucket() {
 	}
 	fmt.Println()
 }
+
+func (self *LogSummaryHandler) dumpBucketJson() {
+	var filters []LogFilter
+	for k := range self.bucketMatches {
+		if !stringz.Contains(self.ignore, k.Id()) {
+			filters = append(filters, k)
+		}
+	}
+	sort.Slice(filters, func(i, j int) bool {
+		return filters[i].Id() < filters[j].Id()
+	})
+	if len(filters) == 0 && self.unmatched == 0 {
+		return
+	}
+
+	model := make(map[string]interface{})
+	model["timestamp"] = self.currentBucket.Format(time.RFC3339)
+	for _, filter := range filters {
+		model[filter.Id()] = self.bucketMatches[filter]
+	}
+	if self.unmatched > 0 {
+		model["unmatched"] = fmt.Sprintf("    unmatched: %0000v\n", self.unmatched)
+	}
+
+	j, err := json.Marshal(model)
+	if err != nil {
```

Comment:
I think it's ok just to panic here. It should never happen, and if it does, there's nothing we can do about it